digital_image_processing/IndexCalculation.py

```diff
@@ -0,0 +1,651 @@
+                #"CIgreen"          --  green, nir
+                #"CIrededge"        --  redEdge, nir
+                #"CI"               --  red, blue
+                #"CTVI"             --  red, nir
+                #"GDVI"             --  green, nir
+                #"EVI"              --  red, blue, nir
+                #"GEMI"             --  red, nir
+                #"GOSAVI"           --  green, nir
+                #"GSAVI"            --  green, nir
+                #"Hue"              --  red, green, blue
+                #"IVI"              --  red, nir
+                #"IPVI"             --  red, nir
+                #"I"                --  red, green, blue
+                #"RVI"              --  red, nir
+                #"MRVI"             --  red, nir
+                #"MSAVI"            --  red, nir
+                #"NormG"            --  red, green, nir
+                #"NormNIR"          --  red, green, nir
+                #"NormR"            --  red, green, nir
+                #"NGRDI"            --  red, green
+                #"RI"               --  red, green
+                #"S"                --  red, green, blue
+                #"IF"               --  red, green, blue
+                #"DVI"              --  red, nir
+                #"TVI"              --  red, nir
+                #"NDRE"               --  redEdge, nir
+
+        #list of all index implemented
+            #allIndex = ["ARVI2", "CCCI", "CVI", "GLI", "NDVI", "BNDVI", "redEdgeNDVI", "GNDVI", "GBNDVI", "GRNDVI", "RBNDVI", "PNDVI", "ATSAVI", "BWDRVI", "CIgreen", "CIrededge", "CI", "CTVI", "GDVI", "EVI", "GEMI", "GOSAVI", "GSAVI", "Hue", "IVI", "IPVI", "I", "RVI", "MRVI", "MSAVI", "NormG", "NormNIR", "NormR", "NGRDI", "RI", "S", "IF", "DVI", "TVI", "NDRE"]
+        #list of index with not blue channel
+            #notBlueIndex = ["ARVI2", "CCCI", "CVI", "NDVI", "redEdgeNDVI", "GNDVI", "GRNDVI", "ATSAVI", "CIgreen", "CIrededge", "CTVI", "GDVI", "GEMI", "GOSAVI", "GSAVI", "IVI", "IPVI", "RVI", "MRVI", "MSAVI", "NormG", "NormNIR", "NormR", "NGRDI", "RI", "DVI", "TVI", "NDRE"]
+        #list of index just with RGB channels
+            #RGBIndex = ["GLI", "CI", "Hue", "I", "NGRDI", "RI", "S", "IF"]
+    """
+
+    def __init__(self, red=None, green=None, blue=None, redEdge=None, nir=None):
+        print("Numpy version: " + np.__version__)
+        self.red = red
+        self.green = green
+        self.blue = blue
+        self.redEdge = redEdge
+        self.nir = nir
+
+    def setMatrices(self, red=None, green=None, blue=None, redEdge=None, nir=None):
+        self.red = red
+        self.green = green
+        self.blue = blue
+        self.redEdge = redEdge
+        self.nir = nir
+        return True
+
+    def calculation(self, index=""):
+        """
+            performs the calculation of the index with the values instantiated in the class
+            :str index: sigla do indice a ser calculado
+        """
+        if index == "":

```

Comment:
Each of these methods already has access to __self__ so they can get their parameters from there.  This would allow you to:
```python
funcs = {
    "ARVI2": self.ARVI2,
    "CCCI": self.CCCI,
    "CVI": self.CVI,
}
try:
    return funcs[index]()
except KeyError:
    print("Index not in the list!")
    return False
```